text/0101-new-scale-control-model-based-on-IO-size.md

```diff
@@ -0,0 +1,82 @@
+## Motivation
+When the difference between the different region's size becomes bigger. It can bring following costs:
+- There is a certain relationship between the speed and size of replica migration. By limiting the number of operators to control the replica speed. It cannot adapt to different region sizes, and cannot associated with IO .
+- The store limit only controls the storage of the source and target instance. In fact, the speed of the operator is mainly affected by the sender and the receiver, so it may cause a serious imbalance in the sending tasks of the individual nodes.
+- The TIKV executes snapshot tasks by FIFO. When the consumption speed of the sender is lower than that of the operator producer, it will take unnecessary wait duration, thereby affecting the timeliness of the operator. So the task will be held longer and other scheduler cannot scheduler this region. 
+Hence, I propose using a new flow mechanism to control the concurrent flow of the sender. It will be helpful to solve the above problems.
+Detailed Design
+The key point of the problem is to balance the speed between the operator producers and the operator consumers. And the operator consumer can be divided into the snapshot sender (region leader ) and receiver (new peer). PD can determine the operator's type and quantity according to the task execution situation. 
+
+## Flow Mechanism
+In order to prevent PD from over-delivering snapshot tasks to the TIKV instance. Each store will have sliding windows to send snapshots, and the total size of the sliding window can be adjusted by the snapshot execution duration and the operator execution. 
+```mermaid
+graph LR
+    subgraph PD
+            subgraph Store
+                RS[regions]
+                SW[sliding window]
+            end
+      SC[Scheduler]-->|1. pick region|RS
+      RS -->|2. check store|SW
+            RS -->|3. region available|SC 
+      SC -->|4. put operator and take token| OC[operator controller]    
+    end 
+
+    
+    
+    subgraph TIKV-leader
+      OC -.->|5. region heartbeat| R[Raft]
+      R -.->|6. generate task| P[pool]
+      P -->|FIFO|P
+      P -.->|7. send task| SE[sender]
+            P -->|7.2 generate duration +|Z
+            SE-->|8.2 send duration +|Z[collect]
+      
+      Z -.->|9.2 store heartbeat|SW
+    end 
+    
+    subgraph TIKV-learner
+        SE -->|8. send snapshot|RE[receiver]
+     RE-->|9. apply snapshot|R1[raft]
+    end
+```
+
+The mechanism of the scheduling limiter is shown in the figure. All scheduler will check the region that's leader has some available send size. If not , it will try another region as target.
+
+### How to adjust windows size 
+The minimum(initial) value of the sliding window is 1000MB. The store can be considered to be scheduled if it's available sending size is bigger than 100MB, so the biggest region can also be scheduled. 
```

Comment:
Is there any lower bound and upper bound?

Reply:
yes, there is lower bound like (1000) , but not upper bound. The increasing size condition is that the wait duration is short than the snapshot executing time and the windows size is not enough.